parsec/mca/termdet/termdet.h

```diff
@@ -0,0 +1,359 @@
+typedef struct parsec_termdet_base_component_2_0_0 parsec_termdet_base_component_t;
+
+/**
+ * @brief possible values for a taskpool state
+ *
+ * @details
+ *   PARSEC_TERM_TP_NOT_MONITORED: the taskpool is not monitored
+ *   PARSEC_TERM_TP_NOT_READY: the taskpool is not ready, the user may still be
+ *       registering work with it
+ *   PARSEC_TERM_TP_BUSY: the taskpool is busy, there is still local work to
+ *       process on it
+ *   PARSEC_TERM_TP_IDLE: the taskpool is idle, termination has not been
+ *       detected, but there is no known local work to process
+ *   PARSEC_TERM_TP_TERMINATED: the global termination has been detected on
+ *       that taskpool.
+ *
+ *  parsec_termination_monitor_taskpool:
+ *    Transitions from PARSEC_TERM_TP_NOT_MONITORED to PARSEC_TERM_TP_NOT_READY
+ *             or from PARSEC_TERM_TP_TERMINATED to PARSEC_TERM_TP_NOT_READY
+ *  parsec_termination_taskpool_ready:
+ *    Transitions from PARSEC_TERM_TP_NOT_READY to PARSEC_TERM_TP_BUSY
+ *  parsec_termination_taskpool_idle:
+ *    Transitions from PARSEC_TERM_TP_BUSY to PARSEC_TERM_TP_IDLE
+ *  parsec_termination_incoming_message:
+ *    Transitions from PARSEC_TERM_TP_IDLE to PARSEC_TERM_TP_BUSY
+ *  When global termination is detected:
+ *    Transitions from PARSEC_TERM_TP_IDLE to PARSEC_TERM_TP_TERMINATED
+ */
+typedef enum {
+    PARSEC_TERM_TP_NOT_MONITORED = 0,
+    PARSEC_TERM_TP_NOT_READY,
+    PARSEC_TERM_TP_BUSY,
+    PARSEC_TERM_TP_IDLE,
+    PARSEC_TERM_TP_TERMINATED
+} parsec_termdet_taskpool_state_t ;
+
+/**
+ * For some termination detectors, we provide two versions of the concept of IDLE:
+ *  - if nb_tasks==0, then the runtime actions are for pending messages to be sent.
+ *    Once the operation of the termination detector has noted that some work is
+ *    potentially located in these outgoing messages (e.g. for the Credit Distribution
+ *    Algorithm, once part of the credit has been assigned to each message), these
+ *    actions have no impact on the idleness of the process. So we can declare
+ *    idleness just after nb_tasks==0 and all outgoing messages have been taken
+ *    into account.
+ *  - if nb_tasks==0 && nb_pending_actions == 0, the process is clearly idle for
+ *    the taskpool, so this is a more generic approach.
+ *
+ *  Which version we use needs to be known globally, because the generated code is
+ *  not the same: in one case (undef TERMDET_XP_IDLE_ON_NBTASKS), we count a fake task 
+ *  to prevent the processes to detect termination before all the startup tasks are
+ *  discovered; in the other case (define TERMDET_XP_IDLE_ON_NBTASKS), we count
+ *  a fake pending action to prevent the processes to detect termination before
+ *  all the startup tasks are discovered.
+ *
+ *  The credit distribution algorithm is slightly better with 
+ *  define TERMDET_XP_IDLE_ON_NBTASKS, but this is still experimental and being
+ *  evaluated.
+ */
+#define TERMDET_XP_IDLE_ON_NBTASKS
```

Comment:
Is this still relevant? I don't see any reference to this macro except for the jdf compiler...

Reply:
Removed in 7f511e94970dff